project/src/main/java/com/google/sps/Models/Database.java

```diff
@@ -419,4 +419,28 @@ private static ArrayList getListProperty(Entity entity, String prop) {
     }
     return commentObjects;
   }
+
+  // Deletes a document
+  public static void deleteDocument(String hash) {
+    Query query = new Query("Document").addFilter("hash", Query.FilterOperator.EQUAL, hash);
+    Entity docEntity = getDatastore().prepare(query).asSingleEntity();
+
+    ArrayList<Long> userIDs = getDocumentUsers(hash);
+    for (long userID : userIDs) {
+      Query queryUser = new Query("User").addFilter(
+        "__key__", Query.FilterOperator.EQUAL, KeyFactory.createKey("User", userID));
+      Entity userEntity = getDatastore().prepare(queryUser).asSingleEntity();
+
+      ArrayList<String> docHashes = getListProperty(userEntity, "docHashes");
+      docHashes.remove(new String(hash));
```

Comment:
out of curiosity why are you creating a new String object here?

Reply:
the remove function takes either an index or an object, so since I am deleting a specific thing I need to pass in an object.